src/utils/fileUtils.ts

```diff
@@ -0,0 +1,75 @@
+import * as path from "path";
+import { TextDocument, window, workspace, WorkspaceFolder } from "vscode";
+
+/**
+ * Return the current workspace uri from root
+ *
+ * @param document the opened TextDocument
+ * @returns the path from root of the current workspace
+ */
+export function getWorkspaceUri(document: TextDocument): string {
+  const currentWorkspace: WorkspaceFolder = (document && document.uri) ? workspace.getWorkspaceFolder(document.uri) : undefined;
+  return ((currentWorkspace && currentWorkspace.uri.fsPath) || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri.fsPath));
+}
+
+/**
+ * Return the current workspace absolute uri
+ *
+ * @param document the opened TextDocument
+ * @returns the absolute path of the current workspace
+ */
+export function getAbsoluteWorkspaceUri(document: TextDocument): string {
+  const currentWorkspace: WorkspaceFolder = (document && document.uri) ? workspace.getWorkspaceFolder(document.uri) : undefined;
+  return ((currentWorkspace && currentWorkspace.uri) || (workspace.workspaceFolders && workspace.workspaceFolders[0].uri)).toString();
+}
+
+/**
+ * Retrieve the text document that is open in the editor
+ *
+ * @returns the text document open in the editor
+ */
+export function getCurrentFile(): TextDocument {
```

Comment:
Remove this function

Reply:
This function takes out the vscode import(s) in variableSubstitution.ts, should I still remove it?

Reply:
This function removes the vscode imports in variableSubstitution.ts. Should I still remove it?